src/d3d9/d3d9_device.cpp

```diff
@@ -6328,61 +6337,56 @@ namespace dxvk {
         // We can resolve the hazard by unbinding the RT.
         m_textureSlotTracking.hazardRT |= 1 << samplerIdx;
 
-        if (unlikely(anyColorWrite && shaderWritesToRt)) {
-          // We have to bind the RT, so we need FEEDBACK_LOOP_LAYOUT.
-          m_textureSlotTracking.unresolvableHazardRT |= 1 << samplerIdx;
-        }
+        if (likely(!anyColorWrite || !shaderWritesToRt))
+          continue;
+
+        // We have to bind the RT, so we need FEEDBACK_LOOP_LAYOUT.
+        m_textureSlotTracking.unresolvableHazardRT |= 1 << samplerIdx;
       }
     }
 
     // Only dirty the framebuffer if we need to make changes for a new hazard
-    if (unlikely(m_textureSlotTracking.hazardRT & ~oldHazardMask) != 0) {
+    if (unlikely(m_textureSlotTracking.hazardRT != oldHazardMask
+      || m_textureSlotTracking.unresolvableHazardRT != oldUnresolvableHazardMask)) {
       m_flags.set(D3D9DeviceFlag::DirtyFramebuffer);
     }
   }
 
 
   inline void D3D9DeviceEx::UpdateActiveHazardsDS(uint32_t texMask) {
-    uint32_t oldHazardMask = m_textureSlotTracking.hazardDS;
+    uint32_t oldHazardMask             = m_textureSlotTracking.hazardDS;
+    uint32_t oldUnresolvableHazardMask = m_textureSlotTracking.unresolvableHazardDS;
     m_textureSlotTracking.hazardDS             &= ~texMask;
     m_textureSlotTracking.unresolvableHazardDS &= ~texMask;
```

Comment:
With the changes, any reason we shouldn't just be setting to 0 now? Isn't this what it's effectively doing...

Reply:
Ah yeah, coming back to me now. :)